### PR TITLE
added geocoding functionality server side for google maps

### PR DESCRIPTION
**Pull request by mengzhouli**
added geocoder fiunctionality and code fromt he slides to the bikes show and index 

including gems 
including js
including yield_js
still need to finish browser side for produciton: Dana needs to generate browser keys in googld api

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -22,7 +22,12 @@ gem 'carrierwave', '~> 0.11.2'
 gem 'omniauth-facebook'
 gem 'attachinary', github: 'assembler/attachinary'
 gem 'jquery-fileupload-rails'
+gem "geocoder"
```
Comment (mengzhouli) [inline Gemfile:25]: for converting location from address to lat and long

File: Gemfile.lock [hunk 2]
```diff
@@ -8,6 +8,7 @@ GIT
```
Comment (mengzhouli) [inline Gemfile.lock:8]: this is all added when i bundle installsed

File: db/migrate/20170412124608_add_coordinates_to_bikes.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddCoordinatesToBikes < ActiveRecord::Migration[5.0]
```
Comment (mengzhouli) [inline db/migrate/20170412124608_add_coordinates_to_bikes.rb:1]: needed this for storing lats and longs to the bikes

File: config/initializers/geocoder.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+Geocoder.configure(
+  # Geocoding options
+  # timeout: 3,                 # geocoding service timeout (secs)
+  lookup: :google,            # name of geocoding service (symbol)
+  # language: :en,              # ISO-639 language code
+  use_https: true,           # use HTTPS for lookup requests? (if supported)
+  # http_proxy: nil,            # HTTP proxy server (user:pass@host:port)
+  # https_proxy: nil,           # HTTPS proxy server (user:pass@host:port)
+  api_key: ENV['GOOGLE_MAPS_SERVER_KEY'],   # API key for geocoding service
```
Comment (dalibran) [inline config/initializers/geocoder.rb:9]: didn't we name this variable differently in application.yml?

Comment (dalibran) [reply config/initializers/geocoder.rb:9]: @mengzhouli

**Review by dalibran — APPROVED** [review]

**Pull request merged**